pandas/io/json/json.py

```diff
@@ -320,49 +334,140 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
                                                       encoding=encoding)
-    if isinstance(filepath_or_buffer, compat.string_types):
-        try:
-            exists = os.path.exists(filepath_or_buffer)
-
-        # if the filepath is too long will raise here
-        # 5874
-        except (TypeError, ValueError):
-            exists = False
-
-        if exists:
-            fh, handles = _get_handle(filepath_or_buffer, 'r',
-                                      encoding=encoding)
-            json = fh.read()
-            fh.close()
-        else:
-            json = filepath_or_buffer
-    elif hasattr(filepath_or_buffer, 'read'):
-        json = filepath_or_buffer.read()
+
+    json_reader = JsonReader(
+        filepath_or_buffer, orient=orient, typ=typ, dtype=dtype,
+        convert_axes=convert_axes, convert_dates=convert_dates,
+        keep_default_dates=keep_default_dates, numpy=numpy,
+        precise_float=precise_float, date_unit=date_unit, encoding=encoding,
+        lines=lines, chunksize=chunksize
+    )
+
+    if chunksize:
+        return json_reader
+
     else:
-        json = filepath_or_buffer
+        return json_reader.read()
 
-    if lines:
+
+class JsonReader(BaseIterator):
+    """
+    Reads a JSON document to a pandas object.
+
+    If initialized with ``lines=True`` and ``chunksize``, can be iterated over
+    ``chunksize`` lines at a time.
+    """
+    def __init__(
+        self, filepath_or_buffer, orient, typ, dtype, convert_axes,
+        convert_dates, keep_default_dates, numpy, precise_float, date_unit,
+        encoding, lines, chunksize, raw_json=False
+    ):
+
+        self.path_or_buf = filepath_or_buffer
+        self.orient = orient
+        self.typ = typ
+        self.dtype = dtype
+        self.convert_axes = convert_axes
+        self.convert_dates = convert_dates
+        self.keep_default_dates = keep_default_dates
+        self.numpy = numpy
+        self.precise_float = precise_float
+        self.date_unit = date_unit
+        self.encoding = encoding
+        self.lines = lines
+        self.chunksize = chunksize
+        self.nrows_seen = 0
+        self.raw_json = False
+
+        if isinstance(filepath_or_buffer, compat.string_types):
+            try:
+                exists = os.path.exists(filepath_or_buffer)
+
+            # if the filepath is too long will raise here
+            # 5874
+            except (TypeError, ValueError):
+                exists = False
+
+            if exists:
+                self.data, _ = _get_handle(filepath_or_buffer, 'r',
+                                           encoding=encoding)
+            else:
+                self.raw_json = True
+                self.data = filepath_or_buffer
+        elif hasattr(filepath_or_buffer, 'read'):
+                self.data = filepath_or_buffer
+        else:
+            self.raw_json = True
+            self.data = filepath_or_buffer
+
+        if self.raw_json and lines:
```

Comment:
why are you doing combine lines here?

Reply:
Moved it out of `_init_` and into `read`, adding another conditional.